plugins/modules/yarn.py

```diff
@@ -230,6 +230,15 @@ def _exec(self, args, run_in_check_mode=False, check_rc=True, unsupported_with_g
 
         return None, None
 
+    def _process_yarn_error(self, err):
+        try:
+            # We need to filter for errors, since Yarn warnings are included in stderr
+            for line in err.splitlines():
+                if json.loads(line)['type'] == 'error':
+                    self.module.fail_json(msg=err)
+        except Exception:
+            self.module.fail_json(msg="Unexpected error from Yarn", err=err)
```

Comment:
How about including the error output in the message:
```suggestion
            self.module.fail_json(msg="Unexpected stderr output from Yarn: %s" % err, stderr=err)
```
I also adjusted the return variable from `err` to `stderr` since that's one of the standard return variables.